core/src/types.rs

```diff
@@ -133,3 +133,47 @@ pub struct PayInvoiceResult {
     pub payment_preimage: String,
     pub fees_msat: Option<u64>,
 }
+
+#[derive(Debug, Deserialize, Serialize, JsonSchema)]
+pub enum FeatureActivationStatus {
+    Mandatory,
+    Optional,
+    Unknown,
+}
+
+#[derive(Debug, Deserialize, Serialize, JsonSchema)]
+pub struct InvoiceFeatures {
+    pub payment_secret: FeatureActivationStatus,
+    pub basic_mpp: FeatureActivationStatus,
+    pub option_payment_metadata: FeatureActivationStatus,
+    pub var_onion_optin: FeatureActivationStatus,
+}
+
+#[derive(Debug, Deserialize, Serialize, JsonSchema)]
+
+pub struct RoutingHint {
+    pub hop_ints: Vec<HopHint>,
```

Comment:
```suggestion
    pub hop_hints: Vec<HopHint>,
```
Typo (haven't check other occurences in the code)